Flow.Launcher.Infrastructure/Alphabet.cs

```diff
@@ -19,160 +19,45 @@ public interface IAlphabet
+        public string Translate(string content)
         {
-            if (!_settings.ShouldUsePinyin || string.IsNullOrEmpty(characters))
-            {
-                return Empty2DStringArray;
-            }
-
-            if (!PinyinCache.ContainsKey(characters))
+            if (_settings.ShouldUsePinyin)
             {
-                var allPinyins = new List<string[]>();
-                foreach (var c in characters)
+                if (_pinyinCache.ContainsKey(content))
                 {
-                    var pinyins = PinyinHelper.toHanyuPinyinStringArray(c, Format);
-                    if (pinyins != null)
+                    if (WordsHelper.HasChinese(content))
                     {
-                        var r = pinyins.Distinct().ToArray();
-                        allPinyins.Add(r);
+                        var result = WordsHelper.GetPinyin(content, ";");
+                        result = GetFirstPinyinChar(result) + result.Replace(";", "");
+                        _pinyinCache[content] = result;
+                        return result;
                     }
                     else
                     {
-                        var r = new[] { c.ToString() };
-                        allPinyins.Add(r);
+                        return content;
                     }
                 }
-
-                var combination = allPinyins.Aggregate(Combination).Select(c => c.Split(';')).ToArray();
-                PinyinCache[characters] = combination;
-                return combination;
+                else
+                    return _pinyinCache[content];
             }
             else
             {
-                return PinyinCache[characters];
+                return content;
             }
         }
 
-        public string Acronym(string[] pinyin)
-        {
-            var acronym = string.Join("", pinyin.Select(p => p[0]));
-            return acronym;
-        }
-
-        public bool ContainsChinese(string word)
+        private string GetFirstPinyinChar(string content)
         {
-            if (!_settings.ShouldUsePinyin)
-            {
-                return false;
-            }
-
-            if (word.Length > 40)
-            {
-                //Skip strings that are too long string for Pinyin conversion.
-                return false;
-            }
-
-            var chinese = word.Select(PinyinHelper.toHanyuPinyinStringArray)
-                              .Any(p => p != null);
-            return chinese;
-        }
-
-        private string[] Combination(string[] array1, string[] array2)
-        {
-            if (!_settings.ShouldUsePinyin)
-            {
-                return EmptyStringArray;
-            }
-
-            var combination = (
-                from a1 in array1
-                from a2 in array2
-                select $"{a1};{a2}"
-            ).ToArray();
-            return combination;
+            return string.Concat(content.Split(';').Select(x => x.First()));
```

Comment:
just a question, why is this code only get the first char, do you know what its used for?

Reply:
This is for getting the acronym for pinyin, just like our discussion in the acronym search. That should be removed once we have done the acronym search in fuzzy search.

Reply:
Hmm if this is for acronym search, could we remove this and add it in the other acronym search PR, I have a feeling we going to forget to remove this code.

Reply:
I have done the part of removing this in the acronym branch. I put it here because I think this will be merged earlier so that I won't like to change the behavior since on the past pinyin acronym search is done with this method.

Reply:
Sure as long as this acronym search code is only specific to Pingyin